src/spaces/productspace.jl

```diff
@@ -151,7 +151,7 @@ function blocksectors(P::ProductSpace{S, N}) where {S, N}
     end
     bs = Vector{I}()
     if N == 0
-        push!(bs, unit(I))
+        return collect(allunits(I))
```

Comment:
This way, the `bs` has not been made in vain, and it sill goes through `sort!` at the end.
```suggestion
        append!(bs, allunits(I))
```